phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -488,7 +488,62 @@ public void rollback(PhoenixConnection connection) throws SQLException {
         numRows = 0;
     }
     
+    private Set<Integer> getUncommittedSattementIndexes() {
+    	Set<Integer> result = newHashSet();
+    	for (Map<ImmutableBytesPtr, RowMutationState> rowMutations : mutations.values()) {
+    		for (RowMutationState rowMutationState : rowMutations.values()) {
+    			result.addAll(rowMutationState.getOrderOfStatementsInConnection());
+    		}
+    	}
+    	return result;
+    }
+    
     @Override
     public void close() throws SQLException {
     }
+    
+    public static class RowMutationState {
+        private Map<PColumn,byte[]> columnValues;
+        private Set<Integer> orderOfStatementsInConnection;
+
+        public RowMutationState(@NotNull Map<PColumn,byte[]> columnValues, int orderOfStatementInConnection) {
+            Preconditions.checkNotNull(columnValues);
+
+            this.columnValues = columnValues;
+            this.orderOfStatementsInConnection = Sets.newHashSet(orderOfStatementInConnection);
+        }
+
+        public RowMutationState(@NotNull Map<PColumn,byte[]> columnValues, @NotNull Set<Integer> orderOfStatementsInConnection) {
+            Preconditions.checkNotNull(columnValues);
+            Preconditions.checkNotNull(orderOfStatementsInConnection);
+            
+            this.columnValues = columnValues;
+            this.orderOfStatementsInConnection = orderOfStatementsInConnection;
+        }
+
+        Map<PColumn, byte[]> getColumnValues() {
+            return columnValues;
+        }
+
+        Set<Integer> getOrderOfStatementsInConnection() {
+            return orderOfStatementsInConnection;
+        }
+        
+        void join(RowMutationState newRow) {
+            for (Map.Entry<PColumn,byte[]> valueEntry : newRow.getColumnValues().entrySet()) {
```

Comment:
Can this for loop be replaced with getColumnValues().putAll(newRow.getColumnValues()) ?